documentation/book/assembly-maintenance-time-windows.adoc

```diff
@@ -0,0 +1,32 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration-kafka.adoc
+
+// Save the context of the assembly that is including this one.
+// This is necessary for including assemblies in assemblies.
+// See also the complementary step on the last line of this file.
+
+[id='assembly-maintenance-time-windows-{context}']
+
+= Maintenance time windows for rolling updates
+
+Maintenance time windows allow you to schedule rolling updates of your Kafka and Zookeeper clusters to start at a convenient time.
+During a rolling update, pod instances in a cluster are updated incrementally, with no downtime.
+
+You configure a maintenance time window by entering an array of strings containing one or more {CronExpressions} in the `Kafka.spec.maintenanceTimeWindows` property.
+The cron expressions must be expressed in UTC.
+The following example shows a typical maintenance time window:
+
+[source,yaml,subs="attributes+"]
+----
+# ...
+maintenanceTimeWindows:
+  - "* * 8-10 * * ?"
+  - "* * 14-15 * * ?"
+# ...
+----
+
+NOTE: Currently, maintenance time windows for rolling updates are only respected by the Certificate Authority (CA) certificate renewal process.
+Other processes--for example, a rolling update triggered by configuring a Kafka cluster--do not respect maintenance time windows.
+
```

Comment:
I think this is still a confusing introduction. Firstly, it's only in this NOTE in the _3rd_ paragraph that the scope of this feature is revealed. And I still don't like how the 2nd sentence explains that a rolling update _is_, after the 1st sentence has talked about being able to schedule one.

Reply:
We have changed this part 3 times with differente request changes. It seems not so simple and there isn't a way which likes to all of us.

Reply:
@tombentley - To your first point, let's shift this content up to the main text (in a new 2nd paragraph) and lose the note. Thank you for calling out this unintentional demotion of core information.

To your second point, I thought quite hard about the ordering of those first two sentences. The purpose of this assembly is, first and foremost, to explain how to schedule maintenance time windows for rolling updates. A quick summary of rolling updates is, to my mind, secondary in importance. As set out in the modular docs guidelines, Links to the documentation on rolling updates are contained in the (Assembly) Additional Resources section, after the included procedure.

@ppatierno - touché!

Reply:
Perhaps a clearer definition in that second sentence would help:

> A rolling update updates every pod instance in a cluster incrementally, with no downtime.

Reply:
If I was a reader and I read that "A rolling update updates every pod instance in a cluster incrementally, with no downtime." I might wonder why being able to schedule it at a convenient time was necessary. 

To be honest I think the problem here is we're trying to write an assembly introduction, but we've not got any concepts. So in trying to keep it brief (because it's an intro) we're not actually explaining the feature (and the motivation behind it) properly.

To understand why this is a useful feature, the reader needs to know:

* Certain things, such as CA updates can happen spontaneously, and that these will cause a rolling update
* That a rolling update causes pods to be restarted
* That while the process is designed maintain availability, it will have a performance impact on the cluster.
* Thus there is a risk that this happens at an inconvenient time.
* Other changes in the cluster happen only when the `Kafka` resource is modified, and therefore the onus is on the user to consider the service-level impact when modifying the resource.

That gives the user enough context to understand what this feature is intended to achieve, and decide whether they want to use it.

If @laidan6000 and @ppatierno agree, I'll put together a concept along these lines and commit it to this PR and take it from there.

Reply:
Fine with me. Go ahead.

Reply:
@tombentley - Thank you for the background. Please go ahead!

Reply:
I've added a couple of concepts, one explaining the need for them and the other talking about cron expressions. Can @ppatierno and @laidan6000 review it please.

Reply:
Fine with me